packages/runtime/container-runtime/src/connectionTelemetry.ts

```diff
@@ -63,10 +68,33 @@ class OpPerfTelemetry {
         this.deltaManager.on("disconnect", () => {
             this.opSendTimeForLatencyStatisticsForMsnStatistics = undefined;
             this.clientSequenceNumberForLatencyStatistics = undefined;
+            this.opTimeSittingInInboundQueue = undefined;
+            this.durationSittingInInboundQueue = undefined;
             this.connectionOpSeqNumber = undefined;
             this.firstConnection = false;
         });
 
+        this.deltaManager.outbound.on("push", (messages) => {
+            for (const msg of messages) {
+                if (this.clientSequenceNumberForLatencyStatistics === msg.clientSequenceNumber) {
+                    assert(this.opTimeSittingInInboundQueue === undefined,
+                        "OpTimeSittingInboundQueue should be undefined");
+                    assert(this.durationSittingInInboundQueue === undefined,
+                        "durationSittingInInboundQueue should be undefined");
+                     this.opTimeSittingInInboundQueue = Date.now();
+                }
+            }
+        });
+
+        this.deltaManager.inbound.on("push", (message: ISequencedDocumentMessage) => {
+            if (this.clientId === message.clientId &&
+                this.clientSequenceNumberForLatencyStatistics === message.clientSequenceNumber &&
+                this.opTimeSittingInInboundQueue !== undefined) {
```

Comment:
I'd convert this.opTimeSittingInInboundQueue check into assert. There should be only one op with given clientId and clientSequenceNumber, and that's nice to assert.